### DISCUSSION TITLE
Update hyperparams generated with annotation in WebUI

### DISCUSSION THREAD

**Discussion by gaocegege** [microsoft/nni#3770] [Ideas]
**What would you like to be added**:

I am one of Kubeflow maintainers mainly focus on tf-operator and katib. I found annotations in nni is really helpful for users. While I think it is not good enough. If the user want to update the search space, he/she has to update the annotation. If we could use techs like syntactic analysis, we could let users update hyperparameters in the web UI, then do the corresponding changes during runtime (Update varaibles in Python).

**Why is this needed**:

Ease to use.

**Without this feature, how does current nni work**：

Python annotations like `@nni.variable(nni.randint(upper),name=variable)`

**Components that may involve changes**:

No idea

**Comment by gaocegege** [comment]
Maybe not in runtime, just ahead of time (generate AST, change value and produce new python script according to the user-provided python file )

**Comment by QuanluZhang** [comment]
In the current version, nni supports updating search space through `nnictl`. It works for nni api. For nni annotation, search space can also be updated, but it is a little tricky, because the search space file is generated automatically by nni. To update search space, users have to write a search space file or find the generated search space file and modify, then use `nnictl update searchspace ...` to update search space. BTW, another user also suggested a similar feature: as the search space file is generated automatically, when a user wants to update search space, he or she has to modify the corresponding annotation in code, may not be as easy as modifying the search space file which has all hyperparameters in one file. Maybe combining search space file and annotation is a possible solution, but we have not come up with an elegant design.

WebUI in the current version has not supported updating search space. This will be supported in future release.

**Comment by gaocegege** [comment]
@QuanluZhang Thanks for your reply. I think what we expect is similar: Do not change the code when we want to change the parameters or search spaces.

All approaches work for me. Feel free to close the issue if there is a similar feature plan, thanks.
